test/Identity.Test/Wrappers/BaseRequestValidatorTestWrapper.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System.Security.Claims;
+using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Services;
+using Bit.Core.Auth.Identity;
+using Bit.Core.Auth.Models.Business.Tokenables;
+using Bit.Core.Auth.Repositories;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Bit.Core.Tokens;
+using Bit.Identity.IdentityServer;
+using Duende.IdentityServer.Models;
+using Duende.IdentityServer.Validation;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+
+namespace Bit.Identity.Test.Wrappers;
+
+public class BaseRequestValidationContextFake
+{
+    public ValidatedTokenRequest ValidatedTokenRequest;
+    public CustomValidatorRequestContext CustomValidatorRequestContext;
+    public GrantValidationResult GrantResult;
+    public string? IpAddress { get; set; }
```

Comment:
💡 , I think this can be removed as the `IpAddress` is read off the `CurrentContext` - not the passed in `Context`.